crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -1836,7 +1836,7 @@ mod test {
                 bytes[31 - (i >> 1)] |= 15 << (4 * (i & 1));
             }
             let h = Felt::from_bytes_be(&bytes);
-            let mut res = Vec::new();
+            let mut res = Vec::with_capacity(enc_len(n_nibbles));
```

Comment:
This is the test. No need to to optimize it. Especially when guessing the len require computation